xwiki-commons-tools/xwiki-commons-tool-webjar-node-handlers/src/main/resources/META-INF/plexus/components.xml

```diff
@@ -0,0 +1,118 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+ *
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ *
+-->
+
+<component-set>
+  <components>
+    <component>
+      <role>org.apache.maven.lifecycle.mapping.LifecycleMapping</role>
+      <role-hint>webjar-node</role-hint>
+      <implementation>org.apache.maven.lifecycle.mapping.DefaultLifecycleMapping</implementation>
+      <configuration>
+        <phases>
+          <process-resources>
+            <mojos>
+              <mojo>
+                <goal>com.github.eirslett:frontend-maven-plugin:install-node-and-pnpm</goal>
+                <configuration>
+                  <nodeVersion>v20.11.0</nodeVersion>
+                  <pnpmVersion>8.15.1</pnpmVersion>
+                  <installDirectory>${basedir}/target</installDirectory>
+                </configuration>
+              </mojo>
+              <mojo>
+                <goal>com.github.eirslett:frontend-maven-plugin:pnpm</goal>
+                <configuration>
+                  <arguments>install --frozen-lockfile</arguments>
+                  <workingDirectory>${basedir}/src/main/node</workingDirectory>
+                  <installDirectory>${basedir}/target</installDirectory>
+                </configuration>
+              </mojo>
+            </mojos>
+          </process-resources>
+          <compile>
+            <mojos>
+              <mojo>
+                <goal>com.github.eirslett:frontend-maven-plugin:pnpm</goal>
+                <configuration>
+                  <arguments>exec tsc --noEmit</arguments>
```

Comment:
TODO: It must be possible to skip typescript type checking in case of project without typescript.